NAME: OKPARAJI STANLEY UDODIRI

EMAIL: OKPARAJISTANLEY9@GMAIL.COM

HAMOYE ID: 14548b1b7c81f000

In [3]:
# Importing libraries
import pandas as pd
import numpy as np

In [4]:
# Loading the dataset
dataset = pd.read_csv("FoodBalanceSheets_E_Africa_NOFLAG.csv", encoding = "Latin-1")
dataset.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

In [5]:
# to know the total number of rows and columns we havw in our dataset
dataset.shape

(60943, 12)

In [6]:
# to check the name of the columns
dataset.columns

Index(['Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element',
       'Unit', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'],
      dtype='object')

#### Question 11

In [7]:
dataset['Item'].value_counts()

Milk - Excluding Butter     1262
Eggs                        1142
Miscellaneous                911
Cereals - Excluding Beer     714
Oilcrops                     706
                            ... 
Grand Total                  135
Animal Products              135
Vegetal Products             135
Sugar non-centrifugal        121
Population                    90
Name: Item, Length: 119, dtype: int64

In [8]:
data = dataset[['Item', 'Y2014', 'Y2017']]
data.groupby(['Item']).sum().loc['Animal fats']

Y2014    209460.54
Y2017    269617.53
Name: Animal fats, dtype: float64

#### Question 12

In [9]:
dataset['Y2015'].describe()

count     59395.000000
mean        135.235966
std        1603.403984
min       -3161.000000
25%           0.000000
50%           0.080000
75%           8.460000
max      181137.000000
Name: Y2015, dtype: float64

In [12]:
# finding the mean and also the std to 3 decimal places
print(f"The mean is = {dataset['Y2015'].mean():.3f}")
print(f"also the standard deviation = {dataset['Y2015'].std():.3f}")

The mean is = 135.236
also the standard deviation = 1603.404


#### Question 13

In [14]:
# To get the total number of the missing value
print(dataset['Y2016'].isnull().sum())
# Percentage to 2 decimal places
per = (dataset['Y2016'].isnull().sum()/len(dataset['Y2016']))*100
print(round(per,2))

1535
2.52


#### Question 14

In [15]:
data1 = dataset.corr().loc['Element Code']
data1

Area Code      -0.000209
Item Code      -0.024683
Element Code    1.000000
Y2014           0.024457
Y2015           0.023889
Y2016           0.023444
Y2017           0.024254
Y2018           0.024279
Name: Element Code, dtype: float64

Y2014 = 0.024457 :is the year with the highest correlation with ‘Element Code’

#### Question 15

In [22]:
data2 = dataset[['Element', 'Y2014','Y2015','Y2016','Y2017','Y2018']]
data2 = data2.groupby(['Element']).sum().loc['Import Quantity']
data2

Y2014    274144.48
Y2015    267018.46
Y2016    286582.78
Y2017    294559.09
Y2018    287997.09
Name: Import Quantity, dtype: float64

In [23]:
data2.max()

294559.09

#### Question 16

In [24]:
data3 = dataset[['Element', 'Y2014']]
data3.groupby(['Element']).sum().loc['Production']

Y2014    1931287.75
Name: Production, dtype: float64

#### Question 17

In [28]:
data4 = dataset[['Y2018','Element']]
data4 = data4.groupby('Element').sum()
data4

Y2018
Element                                           
Domestic supply quantity                2161192.10
Export Quantity                          181594.80
Fat supply quantity (g/capita/day)        10258.69
Feed                                     233489.68
Food                                    1303841.28
Food supply (kcal/capita/day)            455261.00
Food supply quantity (kg/capita/yr)       49056.85
Import Quantity                          287997.09
Losses                                   163902.00
Other uses (non-food)                     91300.97
Processing                               308429.00
Production                              2075072.89
Protein supply quantity (g/capita/day)    11833.56
Residuals                                 34864.00
Seed                                      25263.14
Stock Variation                           20577.91
Total Population - Both sexes           1140605.00
Tourist consumption                          90.00

In [29]:
# To get the maximum value
display(data4.max())

# we can also use
display(data4.nlargest(n=1,columns='Y2018'))

Y2018    2161192.1
dtype: float64

Y2018
Element                            
Domestic supply quantity  2161192.1

#### Question 18

In [33]:
data5 = dataset[['Y2018','Element']]
data5 = data5.groupby('Element').sum().sort_values('Y2018')
data5

Y2018
Element                                           
Tourist consumption                          90.00
Fat supply quantity (g/capita/day)        10258.69
Protein supply quantity (g/capita/day)    11833.56
Stock Variation                           20577.91
Seed                                      25263.14
Residuals                                 34864.00
Food supply quantity (kg/capita/yr)       49056.85
Other uses (non-food)                     91300.97
Losses                                   163902.00
Export Quantity                          181594.80
Feed                                     233489.68
Import Quantity                          287997.09
Processing                               308429.00
Food supply (kcal/capita/day)            455261.00
Total Population - Both sexes           1140605.00
Food                                    1303841.28
Production                              2075072.89
Domestic supply quantity                2161192.10

In [35]:
# To get the third minimum value
data5.iloc[2]

Y2018    11833.56
Name: Protein supply quantity (g/capita/day), dtype: float64

In [37]:
# we can also use
data5.nsmallest(n=3,columns='Y2018')

Y2018
Element                                         
Tourist consumption                        90.00
Fat supply quantity (g/capita/day)      10258.69
Protein supply quantity (g/capita/day)  11833.56

#### Question 19

In [38]:
dataset.groupby(['Area', 'Element']).sum().loc['Algeria', 'Import Quantity']['Y2018']

36238.29

#### Question 20

In [39]:
dataset['Area'].nunique()

49

In [40]:
# we can also use this
len(dataset.groupby(['Area'])['Area'].count())

49